# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/PContent.ipdl
# Commit: fde2d124a1b1
# Full Hash: fde2d124a1b15877570390c7c69f5fcf6770640d
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-08 03:36:50
# Regressor Bug: 1616353
# File Overlap Count: 1
# Description:
#   Bug 1616353 - Part 7.2: Create and use nsOpenWindowInfo types in nsWindowWatcher logic, r=kmag
#   
#   This patch builds on top of part 7.1 by creating this object within
#   nsWindowWatcher and ContentParent to carry the relevant information through
#   provider interfaces when creating new content windows. The nsOpenWindowInfo
# ==============================================================================

diff -r 7ccbbb127b7f -r fde2d124a1b1 dom/ipc/PContent.ipdl
--- a/dom/ipc/PContent.ipdl	Tue Apr 07 21:39:24 2020 +0000
+++ b/dom/ipc/PContent.ipdl	Tue Apr 07 21:39:32 2020 +0000
@@ -1363,6 +1363,7 @@
     sync GetHyphDict(nsIURI aURI) returns (Handle aHandle, uint32_t aSize);
 
     async CreateWindow(nullable PBrowser aThisTab,
+                       MaybeDiscardedBrowsingContext aParent,
                        PBrowser aNewTab,
                        uint32_t aChromeFlags,
                        bool aCalledFromJS,
@@ -1372,11 +1373,13 @@
                        float aFullZoom,
                        Principal aTriggeringPrincipal,
                        nsIContentSecurityPolicy aCsp,
-                       nsIReferrerInfo aReferrerInfo)
+                       nsIReferrerInfo aReferrerInfo,
+                       OriginAttributes aOriginAttributes)
         returns (CreatedWindowInfo window);
 
     async CreateWindowInDifferentProcess(
       PBrowser aThisTab,
+      MaybeDiscardedBrowsingContext aParent,
       uint32_t aChromeFlags,
       bool aCalledFromJS,
       bool aWidthSpecified,
@@ -1386,7 +1389,8 @@
       nsString aName,
       nsIPrincipal aTriggeringPrincipal,
       nsIContentSecurityPolicy aCsp,
-      nsIReferrerInfo aReferrerInfo);
+      nsIReferrerInfo aReferrerInfo,
+      OriginAttributes aOriginAttributes);
 
     /**
      * Tells the parent to ungrab the pointer on the default display.